src/hooks/useApp.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  useMemo,
+  useState,
+  useEffect,
+  useCallback,
+} from 'react';
+
+import {
+  addTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from '../api/todos';
+
+import { FilterType } from '../types/FilterType';
+import { ErrorType } from '../types/ErrorType';
+import { Todo } from '../components/TodoItem/Todo';
+import { getFilteredTodos } from '../helpers/helpers';
+
+const USER_ID = 6748;
+
+export const useApp = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<ErrorType>(ErrorType.NONE);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [unableField, setUnableField] = useState(false);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+
+  const showError = (errorType: ErrorType) => {
+    setError(errorType);
+    setTimeout(() => setError(ErrorType.NONE), 3000);
+  };
+
+  const handleCloseError = useCallback(() => {
+    setError(ErrorType.NONE);
+  }, []);
+
+  const activeTodos = useMemo(() => {
+    return todos.filter(todo => !todo.completed);
+  }, [todos]);
+
+  const completedTodos = useMemo(() => {
+    return todos.filter(todo => todo.completed);
+  }, [todos]);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch {
+      showError(ErrorType.LOAD);
+      setTodos([]);
+    }
+  }, []);
+
+  const addNewTodo = useCallback(
```

Comment:
If we talk about custom hooks - it's a good example of what can you move to it, separate function that requires useCallback can be moved to separate hook in order to split your logic from big component to smaller logical parts